doc/source/profiling.md

```diff
@@ -0,0 +1,90 @@
+# Profiling
+
+GeoUtils has a built-in profiling tool, that can be used to provide more insight on the memory and time use of a function if needed.
+```{warning}
+The profiling functionalities rely on psutil and plotly, which can be installed manually or using Geoutils development dependencies.
+```
+
+With the profiling activated with the graphs output, two kinds of .HTML graphs will be created by default :
+* an icicle graph `time_graph.html`, showing the time spent in each step of the entire process
+* a graph `memory_[function].html` for each decorated functions used, showing the memory consumption of xDEM at regular intervals during the execution
+
+## Configuration and parameters
+
+GeoUtils's profiling configuration works just like a pipeline step. It is executed only if at least one of this two parameters is set to True :
+
+| Name              | Description                       | Type | Default value | Required |
+|-------------------|-----------------------------------| ------- | ------- | ------- |
+| **save_graphs**   | Save the default graphs generated | bool | False | No |
+| **save_raw_data** | Save the raw data on calls as a .pickle file | bool | False | No |
+
+Example of initialization:
+
+```{code-cell} ipython3
+from geoutils.profiler import Profiler
+
+Profiler.enable(save_graphs=True, save_raw_data=True)
+```
+
+After this, if `save_graphs` or `save_raw_data` are True, every profiled function will be studied by the profiler.
+
+## Saved profiling data
+
+The function `generate_summary(output)` saves the profiling information in the output path directory given in input (output_profiling if not specified).
+
+- When `save_graphs` is enabled, the graphs as listed as before are exported.
+- when `save_raw_data` is enabled, GeoUtils saves all stored information as a .pickle file containing a {class}`~pandas.DataFrame` with the following structure:
```

Comment:
Why pickling a pandas DataFrame? Why not use a standard ouput format like Parquet?